ipamapi/contract.go

```diff
@@ -71,7 +71,7 @@ type Ipam interface {
 	// ReleasePool releases the address pool identified by the passed id
 	ReleasePool(poolID string) error
 	// Request address from the specified pool ID. Input options or required IP can be passed.
-	RequestAddress(string, net.IP, map[string]string) (*net.IPNet, map[string]string, error)
+	RequestAddress(string, net.IP, map[string]string) (*net.IPNet, []string, []string, map[string]string, error)
```

Comment:
The return parameter `map[string]string` was in fact added to accommodate remote drivers requirements.
In this case it was not added to carry opaque driver data only. Like we do for the one in the input parameter list, where we pass the container's MAC address if available.

Therefore, I would avoid to change the parameter list for this API.